runtime/vm/resolvesupport.cpp

```diff
@@ -872,6 +872,14 @@ resolveInstanceFieldRefInto(J9VMThread *vmStruct, J9Method *method, J9ConstantPo
 		nameAndSig = J9ROMFIELDREF_NAMEANDSIGNATURE(romFieldRef);
 		name = J9ROMNAMEANDSIGNATURE_NAME(nameAndSig);
 		signature = J9ROMNAMEANDSIGNATURE_SIGNATURE(nameAndSig);
+		// if (J9_ARE_ALL_BITS_SET(->classFlags, J9ClassIsFlattened)) {
+		// 	J9FlattenedClassCache *flattenedClassCache = classFromCP->flattenedClassCache;
+		// 	UDATA index = findIndexInFlattenedClassCache(flattenedClassCache, nameAndSig);
+		// 	fieldOffset = J9_VM_FCC_ENTRY_FROM_FCC(flattenedClassCache, index);
+		// 	if (fieldOffset != -1) {
+		// 		goto done;
+		// 	}
+		// }
```

Comment:
I removed it because the comment was incomplete.

You need to do two things
1) In `loadFlattenableFieldValueClasses` instead of saving the NaS save the `field`. The `J9FlattenedClassCacheEntry` structure needs to be updated.

2) If the `resolvedClass` a value type perform the following optimization
      - check if the field is a Qtype and exists in the FCC in the resolved class
      - if so, check if the offset has already been determined
      - if so, set `resolvedClass` to `definingClass`, this works because ValueTypes can not have superTypes with fields
      - set `fieldOffset` the offset in the FCC
      - set `field` to the field value in field in FCC
      - skip the `instanceFieldOffsetWithSourceClass` call
      - save field index in FCC, set to valueOffset, skip the code that sets the FCC

Reply:
@tajila Is it possible that there are two entries in a FCC with the same NaS? eg. flattenedLine2D has 2 Point2D fields. Are those fields having the same NaS in the FCC?

Reply:
I think I have the answer, they have different NaS?

Reply:
@tajila Please review;